Add information about decoding requests using Truffle Dashboard. (#1385)
trufflesuite/trufflesuite.com: src/docs/truffle/how-to/use-the-truffle-dashboard.md

```

## Decode requests

Truffle Dashboard supports decoding requests for certain RPC calls, allowing you to view a more human-friendly

> representation of the messaging between your computer and the Ethereum network.

Reviewer: Somewhere in here it might be worth mentioning explicitly that this feature currently only supports Truffle projects (and maybe also that we are keen to add Hardhat support in the near future.)

Or maybe this note belongs in the "Usage with non-Truffle tooling" section below? Or maybe in both places? I don't know... you're the professional writer :)

Author: Updated in both places

<p class="alert alert-warning">
<i class="far fa-exclamation-triangle"></i> <strong>Warning</strong>: The decode functionality is available
for the following methods: <code>eth_sendTransaction</code>, <code>personal_sign</code>, <code>eth_signTypedData_v3</code>, and
<code>eth_signTypedData_v4</code>.
</p>

Each time you run `truffle compile`, Truffle shares information about your contracts with Truffle Dashboard.
Later, when you call a method, Truffle Dashboard consults its repository of this compilation
information, and asks `@truffle/decoder` to translate the request into a human readable format.

For example, assume you have a contract that allows you to mint 5 tokens and send it to a
user (`myaccount.eth` in this example), the low-level form would look something like:

```

> 0xa0e9439c000000000000000000000000efef50ebacd8da3c13932ac204361b704eb8292c0000000000000000000000000000000000000000000000000000000000000005

Reviewer: Let's change this to not include my personal address ;)

```suggestion
0xa0e9439c000000000000000000000000627306090abab3a6e1400e9345bc60c78a8bef570000000000000000000000000000000000000000000000000000000000000005
```

in Truffle projects for the following methods: <code>eth_sendTransaction</code>, <code>personal_sign</code>, <code>eth_signTypedData_v3</code>, and
<code>eth_signTypedData_v4</code>. We plan to support non-Truffle tooling such as Hardhat in the future.
</p>

Each time you run `truffle compile`, Truffle shares information about your contracts with Truffle Dashboard.
Later, when you call a method, Truffle Dashboard consults its repository of this compilation
information, and asks `@truffle/decoder` to translate the request into a human readable format.

For example, if you have a contract that allows you to mint 5 tokens and send it to a
user (`myaccount.eth` in this example), the low-level form would look something like:

```
0xa0e9439c000000000000000000000000627306090abab3a6e1400e9345bc60c78a8bef570000000000000000000000000000000000000000000000000000000000000005
```

Truffle Dashboard can display this in a more human-readable format, for example:

```
mint(myaccount.eth, 5)
```

## Usage with non-Truffle tooling

To make the Truffle Dashboard accessible to everyone, we developed it to be agnostic about the tools

> you're using. You can use the Truffle Dashboard with non-Truffle tools such as Hardhat.

Reviewer: This paragraph is much better than the existing :)

```

## Decode requests

Truffle Dashboard supports decoding requests for certain RPC calls, allowing you to view a more human-friendly
representation of the messaging between your computer and the Ethereum network.

<p class="alert alert-warning">

> <i class="far fa-exclamation-triangle"></i> <strong>Warning</strong>: The decode functionality is available

Reviewer 2: Unclear why this is labeled as a "warning" instead of a regular "note" or "important" call-out.

in Truffle projects for the following methods: <code>eth_sendTransaction</code>, <code>personal_sign</code>, <code>eth_signTypedData_v3</code>, and

> <code>eth_signTypedData_v4</code>. Support for non-Truffle tooling such as Hardhat may be added in the future.

Reviewer: (Same note here about lack of need for "may")

Later, when you call a method, Truffle Dashboard consults its repository of this compilation
information, and asks `@truffle/decoder` to translate the request into a human readable format.

For example, if you have a contract that allows you to mint 5 tokens and send it to a
user (`myaccount.eth` in this example), the low-level form would look something like:

```
0xa0e9439c000000000000000000000000627306090abab3a6e1400e9345bc60c78a8bef570000000000000000000000000000000000000000000000000000000000000005
```

Truffle Dashboard can display the this in a more human-readable format, for example:

```
mint(myaccount.eth, 5)
```

## Usage with non-Truffle tooling

We know that not everyone uses Truffle for their smart contract development, but we believe that the
Truffle Dashboard should be accessible to everyone. This is why we developed the Truffle Dashboard to
be agnostic about the tools you're using, so it can also be used with other tools such as Hardhat.

<p class="alert alert-info">
<i class="far fa-info-circle"></i> <strong>Information</strong>: Truffle Dashboard cannot decode requests

> when using non-Truffle tooling. Support for tools such as Hardhat may be added in the future.

Reviewer: ```suggestion
when using non-Truffle tooling. Support for tools such as Hardhat is planned for the near future.
```

Reviewer: (Or make this active voice... but there's no need for uncertainty. I'm putting this one into OKRs.)
